hs/app/reach-cli/Main.hs

```diff
@@ -0,0 +1,352 @@
+import System.Posix.IO (stdError)
+
+import qualified Data.Text.Lazy    as T
+import qualified Data.Text.Lazy.IO as T
+
+default (T.Text)
+
+-- TODO usage help currently prints `reach-cli` instead of `reach`, which may
+-- be confusing to end-users who shouldn't have to care that their CLI session
+-- is being forwarded to this binary... Consider renaming
+
+-- TODO
+reachVersion :: T.Text
+reachVersion = "0.1"
+
+reachImages :: [T.Text]
+reachImages =
+  [ "reach"
+  , "ethereum-devnet"
+  , "algorand-devnet"
+  , "devnet-cfx"
+  , "runner"
+  , "react-runner"
+  , "rpc-server"
+  ]
+
+
+data Cli
+  = Clean FilePath T.Text
+  | Compile
+  | Devnet
+  | DockerReset
+  | Down
+  | Hashes
+  | Help
+  | Init
+  | NumericVersion
+  | React
+  | ReactDown
+  | RpcRun
+  | RpcServer
+  | RpcServerDown
+  | Run
+  | Scaffold
+  | Unscaffold
+  | Update
+  | Upgrade
+  | Version
+  | Whoami
+
+
+--------------------------------------------------------------------------------
+cmdClean :: Mod CommandFields Cli
+cmdClean = command "clean" . info opts $ fullDesc <> desc <> fdoc
+ where
+  desc = progDesc "Delete 'build/$MODULE.$IDENT.mjs'"
+  fdoc = footerDoc . Just
+     $  text "MODULE is \"index\" by default"
+   <$$> text "IDENT  is \"main\"  by default"
+   <$$> text ""
+   <$$> text "If:"
+   <$$> text " * MODULE is a directory then `cd $MODULE && rm -f \"build/index.$IDENT.mjs\";"
+   <$$> text " * MODULE is <something-else> then `rm -f \"build/$MODULE.$IDENT.mjs\""
+  opts = Clean
+    <$> strArgument (metavar "MODULE" <> value "index" <> showDefault)
+    <*> strArgument (metavar "IDENT"  <> value "main"  <> showDefault)
+
+
+-- | > rm -f "build/$m.$i.mjs
+clean :: FilePath -> T.Text -> Script ()
+clean m i = do
+  let f m' = cmd "rm" "-f" $ "build/" <> T.pack m' <> "." <> i <> ".mjs"
+  if m == "index"
+    then f m
+    else ifCmd (test $ TDirExists m)
+      (cmd "cd" m -||- cmd "exit" "1" *> f "index")
+      (f m)
+
+
+--------------------------------------------------------------------------------
+cmdCompile :: Mod CommandFields Cli
+cmdCompile = command "compile" $ info (pure Compile) desc where
+  desc = progDesc "Compile an app"
+
+
+compile :: Script ()
+compile = undefined
+
+
+--------------------------------------------------------------------------------
+cmdInit :: Mod CommandFields Cli
+cmdInit = command "init" $ info (pure Init) desc where
+  desc = progDesc "Set up source files for a simple app"
+
+
+init' :: Script ()
+init' = undefined
+
+
+--------------------------------------------------------------------------------
+cmdRun :: Mod CommandFields Cli
+cmdRun = command "run" $ info (pure Run) desc where
+  desc = progDesc "Run a simple app"
+
+
+run' :: Script ()
+run' = undefined
+
+
+--------------------------------------------------------------------------------
+cmdDown :: Mod CommandFields Cli
+cmdDown = command "down" $ info (pure Down) desc where
+  desc = progDesc "Halt any dockerized devnets for this app"
```

Comment:
You should be taking the opportunity to improve things. This should be `Dockerized`